pallets/alt-vr/src/tests/mock.rs

```diff
@@ -0,0 +1,113 @@
+use crate::{self as ternoa_altvr, Config};
+use frame_support::parameter_types;
+use frame_support::traits::{Contains, GenesisBuild};
+use sp_core::H256;
+use sp_runtime::testing::Header;
+use sp_runtime::traits::{BlakeTwo256, IdentityLookup};
+
+type UncheckedExtrinsic = frame_system::mocking::MockUncheckedExtrinsic<Test>;
+type Block = frame_system::mocking::MockBlock<Test>;
+
+pub const ALICE: u64 = 1;
+
+frame_support::construct_runtime!(
+    pub enum Test where
+        Block = Block,
+        NodeBlock = Block,
+        UncheckedExtrinsic = UncheckedExtrinsic,
+    {
+        System: frame_system::{Pallet, Call, Config, Storage, Event<T>},
+        Altvr: ternoa_altvr::{Pallet, Call, Storage, Event<T>},
+    }
+);
+
+pub struct TestBaseCallFilter;
+impl Contains<Call> for TestBaseCallFilter {
+    fn contains(c: &Call) -> bool {
+        match *c {
+            // For benchmarking, this acts as a noop call
+            Call::System(frame_system::Call::remark(..)) => true,
+            // For tests
+            _ => false,
+        }
+    }
+}
+
+parameter_types! {
+    pub const BlockHashCount: u64 = 250;
+    pub BlockWeights: frame_system::limits::BlockWeights =
+        frame_system::limits::BlockWeights::simple_max(1024);
+}
+impl frame_system::Config for Test {
+    type BaseCallFilter = TestBaseCallFilter;
+    type BlockWeights = ();
+    type BlockLength = ();
+    type DbWeight = ();
+    type Origin = Origin;
+    type Index = u64;
+    type BlockNumber = u64;
+    type Call = Call;
+    type Hash = H256;
+    type Hashing = BlakeTwo256;
+    type AccountId = u64;
+    type Lookup = IdentityLookup<Self::AccountId>;
+    type Header = Header;
+    type Event = Event;
+    type BlockHashCount = BlockHashCount;
+    type Version = ();
+    type PalletInfo = PalletInfo;
+    type AccountData = pallet_balances::AccountData<u64>;
+    type OnNewAccount = ();
+    type OnKilledAccount = ();
+    type SystemWeightInfo = ();
+    type SS58Prefix = ();
+    type OnSetCode = ();
+}
+
+parameter_types! {
+    pub const MaxNameLength: u8 = 20;
+    pub const MinNameLength: u8 = 1;
+}
+
+impl Config for Test {
+    type Event = Event;
+    type WeightInfo = ();
+    type MaxNameLength = MaxNameLength;
+    type MinNameLength = MinNameLength;
+}
+
+pub struct ExtBuilder {
+    altvrs: Vec<(u8, Vec<u8>, Vec<u8>)>,
```

Comment:
Instead of `altvrs: Vec<(u8, Vec<u8>, Vec<u8>)>,` you have probably meant altvrs: Vec<(**u64**, Vec<u8>, Vec<u8>)>,

Reply:
done